src/test/munit/xa/activemq-xa-other-broker-test-case.xml

```diff
@@ -0,0 +1,235 @@
+                operation must be committed -->
+        <munit:execution>
+            <try>
+                <try transactionType="XA" transactionalAction="ALWAYS_BEGIN">
+                    <set-payload value="first"/>
+                    <flow-ref name="publish-on-memory"/>
+                    <set-payload value="second"/>
+                    <flow-ref name="publish-on-other-broker-illegal-body"/>
+                    <error-handler>
+                        <on-error-continue type="JMS:ILLEGAL_BODY"/>
+                    </error-handler>
+                </try>
+                <error-handler>
+                    <on-error-continue/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <jms:consume config-ref="JMS_Config_XA" destination="someOtherDest"/>
+            <munit-tools:assert-that expression="#[payload]" is="#[MunitTools::equalTo('first')]"/>
+            <try>
+                <jms:consume config-ref="JMS_Config_XA_OTHER_BROKER" destination="someDest"/>
+                <munit-tools:fail message="Operation should have been rolled back"/>
+                <error-handler>
+                    <on-error-continue type="JMS:TIMEOUT"/>
+                </error-handler>
+            </try>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="multiple-operation-then-fails-and-rollback">
+        <!--    When error happens after multiple operations, during a new operation, and error is handled by an
+                on-error-propagate, the operations applied before must be rolled back -->
+        <munit:execution>
+            <try>
+                <try transactionType="XA" transactionalAction="ALWAYS_BEGIN">
+                    <set-payload value="first"/>
+                    <flow-ref name="publish-on-memory"/>
+                    <set-payload value="second"/>
+                    <flow-ref name="publish-on-other-broker"/>
+                    <set-payload value="third"/>
+                    <flow-ref name="publish-on-memory"/>
+                    <set-payload value="fourth"/>
+                    <flow-ref name="publish-on-memory"/>
+                    <flow-ref name="publish-on-other-broker-illegal-body"/>
+                </try>
+                <error-handler>
+                    <on-error-continue type="JMS:ILLEGAL_BODY"/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <try>
+                <jms:consume config-ref="JMS_Config_XA_OTHER_BROKER" destination="someDest"/>
+                <munit-tools:fail message="Operation should have been rolled back"/>
+                <error-handler>
+                    <on-error-continue type="JMS:TIMEOUT"/>
+                </error-handler>
+            </try>
+            <try>
+                <jms:consume config-ref="JMS_Config_XA" destination="someOtherDest"/>
```

Comment:
the default timeout is 10 seconds, it's a lot of time, reduce it to 1 second for all consume operations